database-plugins/src/test/java/io/cdap/plugin/db/connector/DBConnectorTest.java

```diff
@@ -122,7 +123,7 @@ private void testSample(DBConnector connector) throws IOException {
     //invalid path
     Assert.assertThrows(IllegalArgumentException.class,
                         () -> connector.sample(new MockConnectorContext(new MockConnectorConfigurer()),
-                                               SampleRequest.builder(1).setPath(schema == null ? "a/b/c" : "a/b/c/d")
+                                               SampleRequest.builder(1).setPath(schema == null ? "a/b" : "a/b/c")
```

Comment:
I don't see a change of behavior in the sample() method, did something change there?

Not sure how this test could be passing before

Reply:
since for generic DB plugin, we require the user to put database in connection string, that means the path at most contain two parts : schema and table.  

this is to test invalid path. the old test case was for the case when database is not required in the connection string and had 3 parts for db that doesn't support schema and 4 parts for db that supports schema. so it still passed the test cases.

Reply:
but I don't see any change in this PR that handles the path differently. Unless I'm missing something?

Reply:
it's in previous PR. But I didn' t update the test case. However old test case still pass for new code. Here , for example

`a/b/c/d` is invalid when path contains database and it's still invalid when path doesn't contain database
I updated it to `a/b/c` in this PR so that this path is only invalid when path doesn't contain database

Reply:
ah I see, understood.